src/Microsoft.ML.Data/Transforms/KeyToVector.cs

```diff
@@ -767,7 +767,7 @@ public override SchemaShape GetOutputSchema(SchemaShape inputSchema)
             {
                 if (!inputSchema.TryFindColumn(colInfo.Input, out var col))
                     throw Host.ExceptSchemaMismatch(nameof(inputSchema), "input", colInfo.Input);
-                if ((col.ItemType.GetItemType().RawKind == default) || !(col.ItemType is VectorType || col.ItemType is PrimitiveType))
+                if (!col.ItemType.GetItemType().RawType.IsValidDataKindType() || !(col.ItemType is VectorType || col.ItemType is PrimitiveType))
```

Comment:
>col.ItemType.GetItemType().RawType.IsValidDataKindType() || !(col.ItemType is VectorType || col.ItemType is PrimitiveType)) [](start = 21, length = 123)

Not your code, but this test seems completely wrong. As in, almost nothing about this test looks right, or even sensible. `col.ItemType` could *never* be a `VectorType` in any instance, according to its definition. In fact under the current type scheme, discounting any types a user may create themselves and the image type, I think this will succeed on everything??

Actually wait, the error message below is wrong too... it will just claim the column is missing (you see it is identical to the test below)... Many of the metadata tests seem off... And of course it claims that it always produces a vector of known size, but that's definitely not true. Wow. I was going to suggest you fix this when I thought it was just confined to one test, but it seems like literally half the lines in this are flat out wrong. OK. :) I'll file a separate issue...

Filed as #2185. Don't bother fixing of course. :D :D